### PR TITLE
deb: add Ubuntu 24.04 (Noble Numbat)

### PR DESCRIPTION
**Pull request by kenhys**

### CODE REVIEW

**Comment by kenhys** [comment]
NOTE: 

```
16.52 
16.52 The following packages have unmet dependencies:
16.58  cmake : Depends: cmake-data (= 3.28.3-1build5) but it is not going to be installed
16.59 E: Unable to correct problems, you have held broken packages.
```

arm64v8/ubuntu:noble cause an  error. 
I'll take a look it later.

**Comment by kenhys** [comment]
> arm64v8/ubuntu:noble cause an error.

It seems that this issue was resolved.

**Comment by kenhys** [comment]
memo: missing ubuntu:24.04 lxc image was blocker before, check it again.

**Comment by kenhys** [comment]
It is strange that some container images were not found...

**Comment by kenhys** [comment]
still ubuntu:24.04 image is not published yet.

**Comment by kenhys** [comment]
It is very weird, the container images for CentOS/7, AmazonLinux 2 and AmazonLinux 2023 - CGroup V1 platform
were not available.

Thus, from ubuntu-20.04 host, these container can't be launched.

Here is the log on ubuntu-20.04 host:

```
lxc image list images:
Run test: list images
  To start your first container, try: lxc launch ubuntu:20.04
  Or for a virtual machine: lxc launch ubuntu:20.04 --vm
  
  +-------+-------------+--------+-------------+--------------+------+------+-------------+
  | ALIAS | FINGERPRINT | PUBLIC | DESCRIPTION | ARCHITECTURE | TYPE | SIZE | UPLOAD DATE |
  +-------+-------------+--------+-------------+--------------+------+------+-------------+
  + echo ::endgroup::
```

Empty image lists.

In contrast to ubuntu-20.04 host, centos/7 images are listed on ubuntu-latest (CGroup V2).
(centos/7 requires cgroup v1, but it was listed on it )

```
  +------------------------------------------+--------------+--------+-------------------------------------------+--------------+-----------------+------------+-------------------------------+
  | centos/7 (3 more)                        | a6930d4481f8 | yes    | CentOS 7 amd64 (20240531_0019)            | x86_64       | VIRTUAL-MACHINE | 405.45MiB  | May 31, 2024 at 12:00am (UTC) |
  +------------------------------------------+--------------+--------+-------------------------------------------+--------------+-----------------+------------+-------------------------------+
  | centos/7 (3 more)                        | f4c23689f2c8 | yes    | CentOS 7 amd64 (20240531_0019)            | x86_64       | CONTAINER       | 85.45MiB   | May 31, 2024 at 12:00am (UTC) |
  +------------------------------------------+--------------+--------+-------------------------------------------+--------------+-----------------+------------+-------------------------------+
  | centos/7/cloud (1 more)                  | 061551e59b88 | yes    | CentOS 7 amd64 (20240531_0019)            | x86_64       | CONTAINER       | 99.48MiB   | May 31, 2024 at 12:00am (UTC) |
  +------------------------------------------+--------------+--------+-------------------------------------------+--------------+-----------------+------------+-------------------------------+
  | centos/7/cloud (1 more)                  | d745955c6b5c | yes    | CentOS 7 amd64 (20240531_0019)            | x86_64       | VIRTUAL-MACHINE | 417.75MiB  | May 31, 2024 at 12:00am (UTC) |
```

lxc image info images:centos/7 on ubuntu-latest shows:

```
Fingerprint: f4c23689f2c89ca2c6d6088e7fb8c02b34fb3b39700d757e3844041562c3df25
  Size: 85.45MiB
  Architecture: x86_64
  Type: container
  Public: yes
  Timestamps:
      Created: 2024/05/31 00:00 UTC
      Uploaded: 2024/05/31 00:00 UTC
      Expires: never
      Last used: never
  Properties:
      architecture: amd64
      variant: default
      os: CentOS
      release: 7
      type: squashfs
      serial: 20240531_0019
      description: CentOS 7 amd64 (20240531_0019)
      requirements.cgroup: v1
  Aliases:
      - centos/7/default
      - centos/7/default/amd64
      - centos/7
      - centos/7/amd64
  Cached: no
  Auto update: disabled
  Profiles: []
```

It seems that ubuntu 20.04 host lxc images seems completely broken.

**Comment by kenhys** [comment]
As #646 was resolved, rebased with master.

File: fluent-package/apt/pkgsize-test.sh [hunk 1]
```diff
@@ -59,6 +62,11 @@ case ${DISTRIBUTION} in
 esac
 
 set -e
+if [ $PREV_DEB_FOUND -ne 1 ]; then
+	echo "The previous version of deb package is not shipped for ${code_name}, so package size check for ${code_name} is disabled"
+	exit 0
+fi
+
```
Comment (daipom) [inline fluent-package/apt/pkgsize-test.sh:69]: How about just outputting the new package size if the `$BASE_NAME` file does not exist?
(Looks like adding a new flag `PREV_DEB_FOUND` is not necessary.)
For example, as follows.

```bash

set -e

DEB=$(find $REPOSITORIES_DIR/${DISTRIBUTION}/pool/${CODE_NAME}/${CHANNEL}/f/fluent-package/fluent-package_*${ARCH}.deb | sort -n | tail -1)
CURRENT_SIZE=$(stat -c %s $DEB)
CURRENT_SIZE_MIB=$(echo "scale=2; ${CURRENT_SIZE} / 1024 / 1024" | bc)

if [ ! -e $BASE_NAME ]; then
    echo "OLD: Not found"
    echo "NEW: ${CURRENT_SIZE_MIB} MiB (${CURRENT_SIZE}) : ${DEB}"
    exit 0
fi

PREVIOUS_SIZE=$(stat -c %s $BASE_NAME)
THRESHOLD_SIZE=`echo "$PREVIOUS_SIZE * 1.2" | bc -l | cut -d. -f1`
PREVIOUS_SIZE_MIB=$(echo "scale=2; ${PREVIOUS_SIZE} / 1024 / 1024" | bc)

echo "OLD: ${PREVIOUS_SIZE_MIB} MiB (${PREVIOUS_SIZE}) : ${BASE_NAME}"
echo "NEW: ${CURRENT_SIZE_MIB} MiB (${CURRENT_SIZE}) : ${DEB}"
if [ $CURRENT_SIZE -gt $THRESHOLD_SIZE ]; then
    echo "${DEB} size (${CURRENT_SIZE}) exceeds ${THRESHOLD_SIZE}. Check whether needless file is bundled or not"
    exit 1
fi
```

File: fluent-package/convert-artifacts-layout.sh
Comment (daipom) [inline fluent-package/convert-artifacts-layout.sh:1]: Don't we need to add `noble` here as well?

https://github.com/fluent/fluent-package-builder/blob/275620daafb0d4d910b2e22aff25087c80633bcb/fluent-package/convert-artifacts-layout.sh#L30

**Review by daipom — COMMENTED** [review]
Thanks!
I have commented on some points.

Also, I have found some code that looks like we should add `noble`:

https://github.com/fluent/fluent-package-builder/blob/275620daafb0d4d910b2e22aff25087c80633bcb/fluent-lts-apt-source/convert-artifacts-layout.sh#L18-L38

https://github.com/fluent/fluent-package-builder/blob/275620daafb0d4d910b2e22aff25087c80633bcb/fluent-package/test-install-in-docker.sh#L85-L104

https://github.com/fluent/fluent-package-builder/blob/275620daafb0d4d910b2e22aff25087c80633bcb/fluent-package/test-verify-repo.sh#L45-L47

File: fluent-package/apt/pkgsize-test.sh (hunk 1 above)
Comment (kenhys) [reply fluent-package/apt/pkgsize-test.sh]: Thanks, it seems better.

**Comment by kenhys** [comment]
Fixed now.

File: fluent-package/apt/pkgsize-test.sh [hunk 2]
```diff
@@ -34,9 +34,6 @@ case ${DISTRIBUTION} in
 	    PREVIOUS_DEB=${BASE_URI}/pool/contrib/f/fluent-package/${BASE_NAME}
 	    set +e
 	    wget ${PREVIOUS_DEB}
-	    if [ $? -eq 0 ]; then
-	       break
-	    fi
```
Comment (daipom) [inline fluent-package/apt/pkgsize-test.sh:39]: Is this change intended?

Comment (kenhys) [reply fluent-package/apt/pkgsize-test.sh]: Thanks, it's not intended, fixed now.

**Review by daipom — APPROVED** [review]
LGTM. Thanks!
